sorc/orog_mask_tools.fd/orog.fd/mtnlm7_oclsm.f

```diff
@@ -914,6 +921,11 @@ SUBROUTINE TERSUB(IMN,JMN,IM,JM,NM,NR,NF0,NF1,NW,EFAC,BLAT,
 C
 C     COMPUTE MOUNTAIN DATA : OA OL
 C
+       allocate (IWORK(IM,JM,4))
+       allocate (OA(IM,JM,4),OL(IM,JM,4),HPRIME(IM,JM,14))
```

Comment:
Are these deallocated?

Reply:
These variables (were previously declared/allocated at the beginning of the subroutine, and then deallocated further down (line 1050). I simply changed them to be allocated just before they were needed rather than at the start of the subroutine

Reply:
Sorry for the deleted response, I began replying and realized I was wrong but hit "comment" by accident.

These are used through the end of the subroutine and so deallocating them there saves no memory. I also did not bother deallocating smaller 2d and 1d arrays; I believe the fortran standard does allow this as variables are deallocated once they are out of scope (such as at the end of a subroutine). But for completeness and consistency I can change all allocatable variables to be explicitly deallocated.